core/src/test/scala/unit/kafka/server/OffsetFetchRequestTest.scala

```diff
@@ -150,42 +149,6 @@ class OffsetFetchRequestTest extends BaseRequestTest {
     }
   }
 
-  @Test
-  def testOffsetFetchRequestWithMultipleGroupsWithOneGroupRepeating(): Unit = {
```

Comment:
I have removed this test because I think that this is wrong. When the same group id is specified multiple times, we should rather throw an invalid request rather than doing this. Thoughts?

This was basically due to the fact that we were using an hash map so the last occurrence of a group won. We don't use the hashmap anymore with the patch. Instead, we return an entry for every group present in the request.

Reply:
Was this behavior documented or just a quirk of the implementation?

I'm also curious if making this change (invalid requests) will break someone's clients.

Reply:
I think that it is a quirk of the implementation. It is because we used HashMaps before this patch so the last one wins.

Reply:
In my opinion, it is pretty bad to just ignore one group passed in the request like this.

Reply:
Are we deciding to actually throw the error? If so we should document that. 

Otherwise, maybe we can add a comment in the request file that requesting the group multiple times in the request will also give us the response multiple times as of this release.

Reply:
I need to look into other APIs to see how we usually handle this. At a first glance, it seems that we are not consistent.

Reply:
Did we ever get a conclusion here? If so let's just document what we decided.

Reply:
+1 for having duplicate responses if a group is repeated. i don't think this warrants an invalid request but we should conform to the norm if there is one.

Reply:
I have looked at other APIs and we are not consistent, unfortunately. I believe that my current implementation (returning a response for each provided group in the same order) is the right way and that could be considered as a bug. I have asked about this in the KIP-709 discuss thread as well.